File: datasets/C++/Qwen/Algorithm/easy1.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
#include <string>

using namespace std;

int alternate(string s) {
    set<char> unique_chars(s.begin(), s.end());
    vector<char> chars(unique_chars.begin(), unique_chars.end());
    int max_length = 0;

    // Try all pairs of distinct characters
    for (int i = 0; i < chars.size(); ++i) {
        for (int j = i + 1; j < chars.size(); ++j) {
            char a = chars[i];
            char b = chars[j];
            string filtered = "";
            bool valid = true;

            for (char c : s) {
                if (c == a || c == b) {
                    filtered += c;
                }
            }

            // Check if alternating
            for (int k = 0; k < filtered.size() - 1; ++k) {
                if (filtered[k] == filtered[k+1]) {
                    valid = false;
                    break;
                }
            }

            if (valid && filtered.length() >= 2) {
                max_length = max(max_length, (int)filtered.length());
            }
        }
    }

    return max_length;
}

int main() {
    int n;
    string s;
    cin >> n;
    cin.ignore(); // ignore newline
    getline(cin, s);

    cout << alternate(s) << endl;
    return 0;
}

File: datasets/Python/Qwen/Algorithm/easy1.py
from itertools import combinations

def alternate(s):
    # Get unique characters
    chars = list(set(s))
    max_length = 0

    # Try all pairs of distinct characters
    for pair in combinations(chars, 2):
        # Filter string to only include characters in this pair
        filtered = [c for c in s if c in pair]
        
        # Check if the filtered string has alternating characters
        valid = True
        for i in range(len(filtered) - 1):
            if filtered[i] == filtered[i+1]:
                valid = False
                break
        
        if valid and len(filtered) >= 2:
            max_length = max(max_length, len(filtered))
    
    return max_length

# Input
n = int(input().strip())
s = input().strip()

# Output
print(alternate(s))